tests/acceptance/test_v2_measurement_service.py

```diff
@@ -0,0 +1,187 @@
+        configuration_parameters=_get_configuration_parameters(
+            name, num_responses, data_size, cumulative_data, response_interval_in_ms, error_on_index
+        )
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    expected_data: List[int] = []
+    index = 0
+    for response in response_iterator:
+        expected_data.extend(index for i in range(data_size))
+        expected = _get_serialized_measurement_outputs(name, index, expected_data)
+        assert expected == response.outputs.value
+        index += 1
+
+    assert index == num_responses
+
+
+@pytest.mark.parametrize("data_size", [1, 10, 100])
+def test___streaming_measurement_service___specify_data_size___receives_expected_amount_of_data(
+    data_size: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    """End to End Test validating measurement returns the expected amount of non-cumulative data."""
+    name = "test"
+    num_responses = 10
+    cumulative_data = False
+    response_interval_in_ms = 1
+    error_on_index = -1
+
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(
+            name, num_responses, data_size, cumulative_data, response_interval_in_ms, error_on_index
+        )
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    index = 0
+    for response in response_iterator:
+        expected_data = [index for i in range(data_size)]
+        expected = _get_serialized_measurement_outputs("test", index, expected_data)
+        assert expected == response.outputs.value
+        index += 1
+
+    assert index == num_responses
+
+
+@pytest.mark.parametrize("error_on_index", [1, 5, 9])
+def test___streaming_measurement_service___specify_error_index___errors_at_expected_response(
+    error_on_index: int, stub_v2: v2_measurement_service_pb2_grpc.MeasurementServiceStub
+):
+    """End to End Test validating streaming measurement sends responses prior to expected error."""
+    name = "test"
+    num_responses = 10
+    data_size = 1
+    cumulative_data = True
+    response_interval_in_ms = 1
+    request = v2_measurement_service_pb2.MeasureRequest(
+        configuration_parameters=_get_configuration_parameters(
+            name, num_responses, data_size, cumulative_data, response_interval_in_ms, error_on_index
+        )
+    )
+
+    response_iterator = stub_v2.Measure(request)
+
+    index = 0
+    with pytest.raises(grpc.RpcError):
+        for response in response_iterator:
+            index += 1
+
+    assert index == error_on_index
+
+
+def _get_configuration_parameters(
+    name: str,
+    num_responses: int,
+    data_size: int,
+    cumulative_data: bool,
+    response_interval_in_ms: int,
+    error_on_index: int,
+) -> any_pb2.Any:
+    serialized_parameter = _get_serialized_measurement_configuration_parameters(
+        name, num_responses, data_size, cumulative_data, response_interval_in_ms, error_on_index
+    )
```

Comment:
You can forward function arguments with `*args` and `**kwargs`.
```suggestion
def _get_configuration_parameters(*args, **kwargs) -> any_pb2.Any:
    serialized_parameter = _get_serialized_measurement_configuration_parameters(*args, **kwargs)
```

It's possible to write type annotations using `ParamSpec`, `ParamSpecArgs`, and `ParamSpecKwargs`, but this requires Python 3.10 and is probably tricky to do in a way that supports older versions.

Reply:
Seems like the build doesn't complain if we don't use type annotations in this case.

Reply:
Values without type annotations have type `typing.Any`, which means you can treat them as any type, call any method on them, etc. Not to be confused with `any_pb2.Any` :D